packages/app-config-writer/src/variants-config/utils.ts

```diff
@@ -0,0 +1,139 @@
+ * @returns sap client
+ */
+export function getSapClientFromPackageJson(scripts: Package['scripts']): string | undefined {
+    for (const value of Object.values(scripts!)) {
+        const match = value?.match(/sap-client=(\d{3})/);
+        if (match) {
+            return match[1];
+        }
+    }
+    return undefined;
+}
+
+/**
+ * Returns the UI5 url parameters.
+ * This is needed for the UI5 run time adaptation.
+ *
+ * @param overwritingParams - parameters to be overwritten
+ * @returns - UI5 url parameters
+ */
+export function getUI5UrlParameters(overwritingParams: Record<string, string> = {}): string {
+    const parameters: Record<string, string> = {
+        'fiori-tools-rta-mode': 'true',
+        'sap-ui-rta-skip-flex-validation': 'true',
+        'sap-ui-xx-condense-changes': 'true'
+    };
+    return stringify(Object.assign(parameters, overwritingParams));
+}
+
+/**
+ * Returns the RTA mount point of the preview middleware configuration from the ui5.yaml file, if given.
+ *
+ * @param previewMiddlewareConfig - configuration of the preview middleware
+ * @returns - RTA mount point or undefined
+ */
+function getRTAMountPoint(previewMiddlewareConfig: PreviewConfigOptions | undefined): string | undefined {
+    if (!isFioriToolsDeprecatedPreviewConfig(previewMiddlewareConfig) && previewMiddlewareConfig?.rta?.editors) {
+        const editors = previewMiddlewareConfig.rta.editors;
+        for (const editor of editors) {
+            if (!('developerMode' in editor)) {
+                return editor.path;
+            }
+        }
+    }
+    return undefined;
+}
+
+/**
+ * Returns the intent of the preview middleware configuration from the ui5.yaml file, if given.
+ *
+ * @param previewMiddlewareConfig - configuration of the preview middleware
+ * @returns - preview intent or undefined
+ */
+function getRTAIntent(previewMiddlewareConfig: PreviewConfigOptions | undefined): string | undefined {
+    if (isFioriToolsDeprecatedPreviewConfig(previewMiddlewareConfig)) {
+        return undefined;
+    }
+    const intent = previewMiddlewareConfig?.flp?.intent;
+    return intent ? `#${intent.object}-${intent.action}` : undefined;
+}
+
+/**
+ * Returns the url for variants management in RTA mode.
+ * The url consist of a specified mount point and intent given from the ui5.yaml file as well as parameters for the RTA mode.
+ *
+ * @param basePath - path to project root, where package.json and ui5.yaml is located
+ * @param query - query to create fragment
+ * @returns - review url parameters
+ */
+export async function getRTAUrl(basePath: string, query: string): Promise<string | undefined> {
+    let previewMiddleware: CustomMiddleware<PreviewConfigOptions> | undefined;
+    try {
+        previewMiddleware = await getPreviewMiddleware(undefined, basePath);
+    } catch (error) {
+        //todo: what to do in case there is no ui5.yaml file? try FileName.Ui5MockYaml or FileName.Ui5LocalYaml as fallback?
```

Comment:
@tobiasqueck you thoughts about potential fallbacks for RTA editor path?

Reply:
While the `ui5.yaml` is not required, it is in every project we generate, so if you are in a project where it doesn't exist, then either something is wrong or the owner knows exactly what they do, therefore, I am ok if our helper tooling doesn't work and they are forced to manually add the script.
So, maybe, just stop here, print an error explaining that we need the `ui5.yaml` to continue or they can add the script manually as described at https://github.com/SAP/open-ux-tools/tree/main/packages/preview-middleware#runtime-adaptation-support

Reply:
Even better: we can use ui5.yaml as default but expert users can provide an other yaml file (path) directly via CLI option 🤓

Reply:
Done with 0810182939708d4afbfa5d85b1f9441bb1d4793d